src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -64,13 +62,28 @@ internal sealed class MultiServerCluster : Cluster, IDnsMonitoringCluster
             : base(settings, serverFactory, eventSubscriber)
         {
             Ensure.IsGreaterThanZero(settings.EndPoints.Count, "settings.EndPoints.Count");
-            if (settings.ConnectionMode == ClusterConnectionMode.Standalone)
+
+#pragma warning disable CS0618
+            if (settings.ConnectionModeSwitch != ConnectionModeSwitch.UseDirectConnection)
             {
-                throw new ArgumentException("ClusterConnectionMode.StandAlone is not supported for a MultiServerCluster.");
+                if (settings.ConnectionMode == ClusterConnectionMode.Standalone)
+#pragma warning restore CS0618
+                {
+                    throw new ArgumentException("ClusterConnectionMode.StandAlone is not supported for a MultiServerCluster.");
+                }
+#pragma warning disable CS0618
+                if (settings.ConnectionMode == ClusterConnectionMode.Direct)
+#pragma warning restore CS0618
+                {
+                    throw new ArgumentException("ClusterConnectionMode.Direct is not supported for a MultiServerCluster.");
+                }
             }
-            if (settings.ConnectionMode == ClusterConnectionMode.Direct)
+            else
             {
-                throw new ArgumentException("ClusterConnectionMode.Direct is not supported for a MultiServerCluster.");
+                if (settings.DirectConnection.GetValueOrDefault())
+                {
+                    throw new ArgumentException("DirectConnection is not supported for a MultiServerCluster.");
+                }
```

Comment:
Replace 66-87 with:

    #pragma warning disable CS0618
                if (settings.ConnectionModeSwitch == ConnectionModeSwitch.UseDirectConnection)
                {
                    if (settings.DirectConnection.GetValueOrDefault())
                    {
                        throw new ArgumentException("DirectConnection is not supported for a MultiServerCluster.");
                    }
                }
                else
                {
                    if (settings.ConnectionMode == ClusterConnectionMode.Standalone)
                    {
                        throw new ArgumentException("ClusterConnectionMode.StandAlone is not supported for a MultiServerCluster.");
                    }
                    if (settings.ConnectionMode == ClusterConnectionMode.Direct)
                    {
                        throw new ArgumentException("ClusterConnectionMode.Direct is not supported for a MultiServerCluster.");
                    }
                }
    #pragma warning restore CS0618

Reply:
From comment in the previous PR the preferred pattern is:

                if (settings.ConnectionModeSwitch == ConnectionModeSwitch.UseDirectConnection)
                {
                   // do the new thing using the new DirectConnection settings
                }
                else
                {
                    // do what we used to do
                }

Note: when there is an `else` clause there is no need for negative tests in the `if` condition.